### PR TITLE
Add tests for no annotation and invalid node annotations

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#1130

### CODE REVIEW

File: manager/integration/tests/test_node.py [hunk 1]
```diff
@@ -1308,3 +1310,150 @@ def test_node_config_annotations(client, core_api,  # NOQA
...
+            "annotations": {
+                DEFAULT_DISK_CONFIG_ANNOTATION:
+                    '[{"path":"/invalid-path","allowScheduling":false,' +
+                    '"storageReserved":1024,"tags":["ssd","fast"]}]',
+                DEFAULT_NODE_TAG_ANNOTATION: '[",.*invalid-tag"]',
+            }
+        }
+    })
+    # Make sure Longhorn has tried to apply the annotation.
+    time.sleep(2)
```
Comment (yasker) [inline manager/integration/tests/test_node.py]: Magic number `2`. Use a constant.

File: manager/integration/tests/test_node.py [hunk 2]
```diff
@@ -1308,3 +1310,150 @@ def test_node_config_annotations(client, core_api,  # NOQA
...
+    # then the correct annotation should be applied.
+    node = client.by_id_node(node_name)
+    disks = node.disks
+    for _, disk in iter(disks.items()):
+        disk.allowScheduling = False
+    update_disks = get_update_disks(disks)
+    node = client.by_id_node(node_name)
+    node.diskUpdate(disks=update_disks)
+    node.diskUpdate(disks=[])
+    time.sleep(2)
```
Comment (yasker) [inline manager/integration/tests/test_node.py]: There is a wait below, why we need to use `time.sleep` here?

File: manager/integration/tests/test_node.py [hunk 3]
```diff
@@ -1308,3 +1310,150 @@ def test_node_config_annotations(client, core_api,  # NOQA
...
+        }
+    })
+    time.sleep(2)
+    node = client.by_id_node(node_name)
+    assert set(node.tags) == {"tag3", "tag4"}
+
+    # Clean up all node tags
+    # then the correct annotation should be applied.
+    client.update(node, tags=[])
+    time.sleep(2)
```
Comment (yasker) [inline manager/integration/tests/test_node.py]: there is a wait below.

File: manager/integration/tests/test_node.py (hunk 2 above)
Comment (shuo-wu) [reply manager/integration/tests/test_node.py]: Before the above disk cleanup, there is one existing disk on the node.
After the disk cleanup, node controller will re-create a new disk immediately according to the annotation.
Hence the below wait cannot guarantee that the cleanup and recreation are complete or not.

Similar below

Comment (yasker) [reply manager/integration/tests/test_node.py]: Then why sleeping 2 seconds can guarantee that?

Comment (shuo-wu) [reply manager/integration/tests/test_node.py]: OK. It's better to use `wait_for_disk_status()` rather than sleep and `wait_for_disk_update`

**Pull request merged**